src/gui/tray/activitydata.cpp

```diff
@@ -44,4 +45,97 @@ ActivityLink ActivityLink::createFomJsonObject(const QJsonObject &obj)
 
     return activityLink;
 }
+
+OCC::Activity Activity::fromActivityJson(const QJsonObject json, const AccountPtr account)
+{
+    const auto activityUser = json.value(QStringLiteral("user")).toString();
+
+    Activity activity;
+    activity._type = Activity::ActivityType;
+    activity._objectType = json.value(QStringLiteral("object_type")).toString();
+    activity._id = json.value(QStringLiteral("activity_id")).toInt();
+    activity._fileAction = json.value(QStringLiteral("type")).toString();
+    activity._accName = account->displayName();
+    activity._subject = json.value(QStringLiteral("subject")).toString();
+    activity._message = json.value(QStringLiteral("message")).toString();
+    activity._file = json.value(QStringLiteral("object_name")).toString();
+    activity._link = QUrl(json.value(QStringLiteral("link")).toString());
+    activity._dateTime = QDateTime::fromString(json.value(QStringLiteral("datetime")).toString(), Qt::ISODate);
+    activity._icon = json.value(QStringLiteral("icon")).toString();
+    activity._isCurrentUserFileActivity = activity._objectType == QStringLiteral("files") && activityUser == account->davUser();
+
+    auto richSubjectData = json.value(QStringLiteral("subject_rich")).toArray();
+
+    if(richSubjectData.size() > 1) {
+        activity._subjectRich = richSubjectData[0].toString();
+        auto parameters = richSubjectData[1].toObject();
+        const QRegularExpression subjectRichParameterRe(QStringLiteral("({[a-zA-Z0-9]*})"));
+        const QRegularExpression subjectRichParameterBracesRe(QStringLiteral("[{}]"));
+
+        for (auto i = parameters.begin(); i != parameters.end(); ++i) {
+            const auto parameterJsonObject = i.value().toObject();
+
+            activity._subjectRichParameters[i.key()] = Activity::RichSubjectParameter  {
+                parameterJsonObject.value(QStringLiteral("type")).toString(),
+                parameterJsonObject.value(QStringLiteral("id")).toString(),
+                parameterJsonObject.value(QStringLiteral("name")).toString(),
+                parameterJsonObject.contains(QStringLiteral("path")) ? parameterJsonObject.value(QStringLiteral("path")).toString() : QString(),
+                parameterJsonObject.contains(QStringLiteral("link")) ? QUrl(parameterJsonObject.value(QStringLiteral("link")).toString()) : QUrl(),
+            };
+        }
+
+        auto displayString = activity._subjectRich;
+        auto i = subjectRichParameterRe.globalMatch(displayString);
```

Comment:
can you improve the name of variable `i` ?

Reply:
Yes, now `subjectRichParameterMatch`